quint/src/effects/modeChecker.ts

```diff
@@ -35,14 +34,10 @@ import { effectToString } from './printing'
  */
 export function checkModes(
   quintModule: QuintModule, effects: Map<bigint, Effect>
-): Either<Map<bigint, ErrorTree>, Map<bigint, OpQualifier>> {
+): [Map<bigint, ErrorTree>, Map<bigint, OpQualifier>] {
```

Comment:
I thought `Either` was our preferred way. Why would we switch to a pair here?

Reply:
In order to return partial information. That is, even if there are errors inferring modes somewhere, the inferred information for definitions where inference was successful can be used. This was an improvement I made for types and effects, but didn't do for modes at the time, so I'm doing it now.